src/features/store/ui/StoreItem.tsx

```diff
@@ -0,0 +1,15 @@
+import type Item from '@type/Item';
+import * as S from './styles';
+import { getImageUrl } from '@utils/getImageUrl';
+
+export default function StoreItem({ id, name, image, category, cost }: Item) {
+  return (
+    <>
+      <S.StoreItem key={id}>
```

Comment:
```suggestion
      <S.StoreItem>
```
이쪽에 `key`가 들어가네요? StoreItem 를 직접 렌더링하는 부모 컴포넌트 측에서 key를 넘기는게 일반적이지 않을까요?
빈 fragment도 제거 될 수 있을 것 같습니다.

```tsx
      <S.ItemContainer $category={query}>
        {testItem.map(item => (
          <StoreItem key={item.id} {...item} />
        ))}
      </S.ItemContainer>
```

하나의 팁으로 빈 Fragment에 key를 주입하고 싶다면 아래와 같은 방법이 있습니다.
```tsx
arr.map((item) => (
  <React.Fragment key={item.id}>
    {/* ... */}
  </React.Fragment>
))
```

Reply:
@ssi02014 
```ts
<S.ItemContainer $category={query}>
        {testItem.map(item => (
          <StoreItem {...item} />
```
원래 반복문 쪽에서 `<StoreItem />` 컴포넌트가 아닌 
```ts
<S.StoreItem>
      <S.ItemLabel>{name}</S.ItemLabel>
      <S.ItemImage $category={category} src={getImageUrl(image)} />
      <S.ItemLabel>{cost} Point</S.ItemLabel>
</S.StoreItem>
```
위 코드같은 태그 덩어리를 반복시켜 주다가 미래를 생각하여 `StoreItem` 컴포넌트 내부로 코드를 옮겼습니다.
그 과정에서 생긴 실수 같네요... https://github.com/modern-agile-team/8term-coko-Front/pull/84#discussion_r1874823641 코멘트도 이와같은 작업을 하다 생긴 일 같습니다 😢

Reply:
@rhehfl 리액트 key prop에 대해서 조금 더 살펴 볼 필요는 있어보여요!
컴포넌트 내부에서 key는 지정 할 필요가 없으며, 리액트에서도 해당 부분에대해서 언급을 하는 내용이 있습니다~
미래를 위한 의도는 이해되지만 개선은 필요해보입니다 🤗

### 잘못된 Key 사용법
```tsx
function ListItem(props) {
  const value = props.value;
  return (
    // 틀렸습니다! 여기에는 key를 지정할 필요가 없습니다.
    <li key={value.toString()}>
      {value}
    </li>
  );
}

function NumberList(props) {
  const numbers = props.numbers;
  const listItems = numbers.map((number) =>
    // 틀렸습니다! 여기에 key를 지정해야 합니다.
    <ListItem value={number} />
  );
  return (
    <ul>
      {listItems}
    </ul>
  );
}
```

### 올바른 Key 사용법
```tsx
function ListItem(props) {
  // 맞습니다! 여기에는 key를 지정할 필요가 없습니다.
  return <li>{props.value}</li>;
}

function NumberList(props) {
  const numbers = props.numbers;
  const listItems = numbers.map((number) =>
    // 맞습니다! 배열 안에 key를 지정해야 합니다.
    <ListItem key={number.toString()} value={number} />
  );
  return (
    <ul>
      {listItems}
    </ul>
  );
```

## 참조
https://ko.legacy.reactjs.org/docs/lists-and-keys.html#extracting-components-with-keys

Reply:
상세한 설명 감사드립니다!
https://github.com/modern-agile-team/8term-coko-Front/pull/84#discussion_r1874924855 코멘트에서 인라인 반복문을 사용해 랜더링을 시키는 부분에 Key를 지정하도록 수정했습니다!

Reply:
덕분에 리액트의 올바른 key 사용법에 대해 알게되었습니다... 
공식문서를 자주 읽는 습관도 기르는 게 좋겠네요!